code/modules/particles/byond_particles/particool.dm

```diff
@@ -0,0 +1,252 @@
+	square         | vector | A random XY-only vector between squares of sizes A and B. (The length of the square is between A*2 and B*2, centered at 0,0.)
+	cube           | vector | A random vector between cubes of sizes A and B. (The length of the cube is between A*2 and B*2, centered at 0,0,0.)
+	*/
+
+	// I write code like this because I hate myself
+	var/a = length(stringToList(L["a"],TRUE)) > 1 ? stringToList(L["a"],TRUE) : text2num(L["a"])
+	var/b = length(stringToList(L["b"],TRUE)) > 1 ? stringToList(L["b"],TRUE) : text2num(L["b"])
+
+
+	var/rand_type = parse_rand_type(L["rand"])
+
+	switch(L["genType"])
+		if("num")    return generator(L["genType"], a, b, rand_type)
+		if("vector") return generator(L["genType"], a, b, rand_type)
+		if("box")    return generator(L["genType"], a, b, rand_type)
+		if("color") //Color can be string or matrix
+			a = length(a) > 1 ? a : L["a"]
+			b = length(b) > 1 ? b : L["b"]
+			return generator(L["genType"], a, b, rand_type)
+		if("circle") return generator(L["genType"], a, b, rand_type)
+		if("sphere") return generator(L["genType"], a, b, rand_type)
+		if("square") return generator(L["genType"], a, b, rand_type)
+		if("cube")   return generator(L["genType"], a, b, rand_type)
+	return null
+
+/datum/particle_editor/proc/parse_rand_type(rand_type)
+	switch(rand_type)
+		TEXT_TO_MACRO(UNIFORM_RAND)
+		TEXT_TO_MACRO(LINEAR_RAND)
+		TEXT_TO_MACRO(NORMAL_RAND)
+		TEXT_TO_MACRO(SQUARE_RAND)
+	CRASH("Unknown rand type [rand_type]")
+
+/datum/particle_editor/proc/debugOutput(L, nodeName)
+	if(istype(L,/list))
+		for(var/elem in L)
+			if(istype(elem,/list))
+				world.log << nodeName
+				debugOutput(elem, nodeName + ":LIST" + elem)
+			else
+				world.log << nodeName + ":" + elem + ":" + text("[]",elem)
+	else
+		world.log << L
+
+/datum/particle_editor/ui_act(action, list/params)
+	. = ..()
+	if(.)
+		return
+
+	debugOutput(params)
+
+	switch(action)
+		if("add_particle")
+			target.add_particle()
+			. = TRUE
+		if("remove_particle")
+			target.remove_particle()
+			. = TRUE
+		if("modify_particle_value")
+			target.modify_particle_value(params["new_data"]["name"], translate_value(params["new_data"]))
+			. = TRUE
+		if("modify_color_value")
+			var/new_color = input(usr, "Pick new particle color", "Particool Colors!") as color|null
+			if(new_color)
+				target.modify_particle_value("color",new_color)
+				. = TRUE
+		if("modify_icon_value")
+			var/icon/new_icon = input("Pick icon:", "Icon") as null|icon
+			if(new_icon && target.particles)
+				target.modify_particle_value("icon", new_icon)
+				. = TRUE
+
+
+//movable procs n stuff
+
+/atom/movable/proc/add_particle()
+	particles = new /particles
+
+/atom/movable/proc/remove_particle()
+	particles = null
+
+/atom/movable/proc/modify_particle_value(varName, varVal)
+	var/list/default_particle = list(width = 100, ///DO NOT make this list look nicer, maintainers at tg didn't like the fact its like this and its funny so keep it :) - Borbop
```

Comment:
what a spiteful bitch
also can you please do the opposite of this it looks like ass, it will require people to scroll right in vscode to see it.

Reply:
didnt notice yeah thats gross af lmao